src/hooks/useWalletConnectConnections.js

```diff
@@ -26,9 +26,28 @@ const walletConnectSelector = createSelector(
   state => state.walletconnect.walletConnectors,
   walletConnectors => {
     const sorted = sortList(values(walletConnectors), 'peerMeta.name');
-    const groupedByDappName = groupBy(sorted, 'peerMeta.url');
+    const groupedByDappName = sorted.reduce((acc, walletConnector) => {
+      if (acc[walletConnector.peerMeta.ur]) {
+        acc[walletConnector.peerMeta.ur].push(walletConnector);
+      } else {
+        Object.assign(acc, {
+          [walletConnector.peerMeta.ur]: [walletConnector],
+        });
+      }
+      return acc;
+    }, {});
     const mostRecent = sortList(sorted, '_handshakeId', 'desc');
-    const sortedByMostRecentHandshake = groupBy(mostRecent, 'peerMeta.url');
+
+    const sortedByMostRecentHandshake = mostRecent.reduce((acc, handshake) => {
+      if (acc[handshake.peerMeta.url]) {
+        acc[handshake.peerMeta.url].push(handshake);
+      } else {
+        Object.assign(acc, {
+          [handshake.peerMeta.url]: [handshake],
+        });
+      }
+      return acc;
+    }, {});
```

Comment:
Maybe we should extract this behaviour to our own `groupBy` function?

From developer perspective it feels like a step backwards, having to write all of this now 😄.